### PR TITLE
BUG: is_categorical shouldnt recognize Dtype objects

### PR DESCRIPTION
**Pull request by jbrockmendel**
is_categorical(obj) is used in a couple places where we then go on to access `obj.dtype`, so this should not return `True` if we have a `CategoricalDtype` object.

### CODE REVIEW

File: pandas/core/dtypes/common.py [hunk 1]
```diff
@@ -68,9 +68,6 @@
 ensure_float64 = algos.ensure_float64
 ensure_float32 = algos.ensure_float32
 
-_ensure_datetime64ns = conversion.ensure_datetime64ns
-_ensure_timedelta64ns = conversion.ensure_timedelta64ns
```
Comment (jbrockmendel) [inline pandas/core/dtypes/common.py:72]: unrelated edit before i figured out what this branch was about; these are never used

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -4716,7 +4716,7 @@ def groupby(self, values) -> PrettyDict[Hashable, np.ndarray]:
         # TODO: if we are a MultiIndex, we can do better
         # that converting to tuples
         if isinstance(values, ABCMultiIndex):
-            values = values.values
+            values = values._values
```
Comment (jbrockmendel) [inline pandas/core/indexes/base.py:4719]: unrelated edit from before i figured out what this branch was about

**Comment by TomAugspurger** [comment]
This will need a deprecation cycle right?

**Comment by jbrockmendel** [comment]
> This will need a deprecation cycle right?

id classify this as a bugfix, but OK either way

**Comment by jbrockmendel** [comment]
the other option is just to deprecate it, since we only use is_categorical in a few places and its more likely than not to cause confusion

**Comment by jbrockmendel** [comment]
Closing in favor of #33385.
